data_model/src/query.rs

```diff
@@ -1520,48 +1515,33 @@ pub mod error {
+            /// Failed to find asset: `{0}`
             Asset(AssetId),
-            /// Failed to find asset definition
-            #[display(fmt = "Failed to find asset definition: `{_0}`")]
+            /// Failed to find asset definition: `{0}`
             AssetDefinition(AssetDefinitionId),
-            /// Failed to find account
-            #[display(fmt = "Failed to find account: `{_0}`")]
+            /// Failed to find account: `{0}`
             Account(AccountId),
-            /// Failed to find domain
-            #[display(fmt = "Failed to find domain: `{_0}`")]
+            /// Failed to find domain: `{0}`
             Domain(DomainId),
-            /// Failed to find metadata key
-            #[display(fmt = "Failed to find metadata key")]
+            /// Failed to find metadata key: `{0}`
             MetadataKey(Name),
-            /// Block with supplied parent hash not found. More description in a string.
-            #[display(fmt = "Block with hash {_0} not found")]
+            /// Block with hash `{0}` not found
             Block(HashOf<VersionedCommittedBlock>),
-            /// Transaction with given hash not found.
-            #[display(fmt = "Transaction not found")]
+            /// Transaction with hash `{0}` not found
             Transaction(HashOf<VersionedSignedTransaction>),
-            /// Peer not found.
-            #[display(fmt = "Peer {_0} not found")]
+            /// Peer with id `{0}` not found
             Peer(PeerId),
-            /// Trigger not found.
-            #[display(fmt = "Failed to find trigger: `{_0}`")]
+            /// Trigger with id `{0}` not found
             Trigger(TriggerId),
-            /// Failed to find Role by id.
-            #[display(fmt = "Failed to find role by id: `{_0}`")]
+            /// Role with id `{0}` not found
             Role(RoleId),
-            /// Failed to find [`PermissionTokenDefinition`] by id.
-            #[display(fmt = "Failed to find permission token definition by id: `{_0}`")]
+            /// Permission token definition with id `{0}` not found
             PermissionTokenDefinition(PermissionTokenId),
-            /// Failed to find [`PermissionToken`] for [`AccountId`].
-            #[cfg_attr(not(feature = "std"), display(fmt = "Failed to find permission token"))]
-            #[cfg_attr(feature = "std", error(transparent))]
+            /// Failed to find permission token
             PermissionToken(PermissionTokenFindError),
-            /// Failed to find specified [`Parameter`] variant.
-            #[display(fmt = "Failed to find specified parameter variant: `{_0}`")]
+            /// Parameter with id `{0}` not found
             Parameter(ParameterId),
-            /// Failed to find [`PublicKey`] at some account.
-            #[display(fmt = "Failed to find public key: `{_0}`")]
+            /// Failed to find [`PublicKey`]: `{0}`
             PublicKey(PublicKey),
```

Comment:
**Inconsistency**: for `::PermissionToken` you deleted doc links (<code>[\`PermissionToken\`]</code>, <code>[\`AccountId\`]</code>), but you preserved it for `::PublicKey` (<code>[\`PublicKey\`]</code>).

So, there are two options (as I see):

1. Remove doc links completely. It would be more consistent with the rest of the code (in other errors you have removed it). 
2. Return doc links for `::PermissionToken`, and possibly in other places as well. Since doc links do not make sense for `displaydoc` (AFAIU), they should be placed below the first doc line.

I would prefer the 1st option. IMO, we should either write docs for `cargo doc` or for `displaydoc`, not for both. Otherwise, it will lead to dirt. With this PR errors' doc strings will be used explicitly for `Display`, not for API documentation, because it uses formatting that doesn't make sense for `cargo doc`.

Reply:
Yeah, the doc links definitely do not make sense for the `displaydoc`

With this in mind, I also lean towards the first option. Losing the links is kinda sad, but if we put them at the second line we would be back to square one with duplication between docs and error messages